payment/src/main/java/org/spine3/samples/lobby/payment/PaymentAggregate.java

```diff
@@ -0,0 +1,160 @@
+            throw new SecondInitializationAttempt(getId());
+        }
+
+        // Update aggregate state
+        final OrderTotal total = command.getTotal();
+        final Money orderCost = total.getTotalPrice();
+        final ThirdPartyProcessorPayment.Builder newState = getState().toBuilder()
+                                                                      .setId(getId())
+                                                                      .setPrice(orderCost);
+        markInitialized(newState);
+
+        // Construct and return events
+        final PaymentInstantiated resultEvent = PaymentInstantiated.newBuilder()
+                                                                   .setId(getId())
+                                                                   .build();
+        return resultEvent;
+    }
+
+    @Assign
+    public Message handle(CompleteThirdPartyProcessorPayment command, CommandContext context) throws FailureThrowable {
+        final boolean successful = command.getSuccessful();
+        final PaymentStatus status = successful ? SUCCEED : FAILED;
+        checkResultStatus(status);
+
+        final ThirdPartyProcessorPayment payment = getState().toBuilder()
+                                                             .setStatus(status)
+                                                             .build();
+        incrementState(payment);
+        final PaymentId id = getId();
+        final Message resultEvent = successful
+                                    ? PaymentCompleted.newBuilder()
+                                                      .setId(id)
+                                                      .build()
+                                    : PaymentRejected.newBuilder()
+                                                     .setId(id)
+                                                     .build();
+        return resultEvent;
+    }
+
+    @Assign
+    public PaymentCanceled handle(CancelThirdPartyProcessorPayment command, CommandContext context) throws FailureThrowable {
+        final PaymentStatus status = CANCELED;
+        checkResultStatus(status);
+
+        final ThirdPartyProcessorPayment payment = getState().toBuilder()
+                                                             .setStatus(status)
+                                                             .build();
+        incrementState(payment);
+        final PaymentId id = getId();
+        final PaymentCanceled resultEvent = PaymentCanceled.newBuilder()
+                                                           .setId(id)
+                                                           .build();
+        return resultEvent;
+    }
+
+    /**
+     * Is the aggregate initialized?
+     *
+     * @return {@code true} if {@link ThirdPartyProcessorPayment#status_ status} is greater or equal to
+     * {@link PaymentStatus#INITIALIZED INITIALIZED}.
+     */
+    public boolean isInitialized() {
```

Comment:
I don't like coupling of the aggregate and the process manager.

Let's draw the architecture on the board.